mm2src/coins/eth/eth_swap_v2.rs

```diff
@@ -96,14 +129,327 @@ impl EthCoin {
                 .compat()
                 .await
             },
-            EthCoinType::Nft { .. } => Err(TransactionErr::ProtocolNotSupported(
-                "NFT protocol is not supported for ETH and ERC20 Swaps".to_string(),
-            )),
+            EthCoinType::Nft { .. } => Err(TransactionErr::ProtocolNotSupported(ERRL!(
+                "NFT protocol is not supported for ETH and ERC20 Swaps"
+            ))),
+        }
+    }
+
+    pub(crate) async fn validate_taker_funding_impl(
+        &self,
+        args: ValidateTakerFundingArgs<'_, Self>,
+    ) -> ValidateSwapV2TxResult {
+        if let EthCoinType::Nft { .. } = self.coin_type {
+            return MmError::err(ValidateSwapV2TxError::Internal(
+                "EthCoinType must be ETH or ERC20".to_string(),
+            ));
+        }
```

Comment:
this doesn't seems to be an `InternalError` but more of `Unsupported...`

Reply:
well we nft feature supports swap v2 maker methods. here we inform that wrong eth coin type was provided

Reply:
Done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/commits/9f3164995bd463227c570ad9bbd890e3103142ec